src/dcat_ap_no_validator_service/service/validator_service.py

```diff
@@ -188,15 +207,26 @@ async def _import_ontologies(self, session: CachedSession) -> None:
             for t in all_imports:
                 self.ontology_graph.remove(t)
             # 3. get all the imported vocabularies and import them
-            for (_s, _p, uri) in all_imports:
-                if (uri, None, None) not in self.data_graph:
-                    if (uri, None, None) not in self.ontology_graph:
-                        logging.debug(f"Trying to fetch remote triples about {uri}.")
-                        try:
-                            _g = await fetch_graph(session, uri)
-                            if _g:
-                                self.ontology_graph += _g
-                        except FetchError:
-                            logging.debug(traceback.format_exc())
-                            pass
+            await asyncio.gather(
+                *[self.fetch_triples(uri, session) for (_s, _p, uri) in all_imports]
+            )
             # 4. start all over again to see if import statements have been imported
+
+    async def fetch_triples(self, uri: str, session: CachedSession) -> None:
+        """Fetch remote triples and add them to the ontology_graph.
+
+        Only triples that are not allready in the data_graph and/or ontology_graph are added.
+        """
+        if (uri, None, None) not in self.data_graph:
+            if (uri, None, None) not in self.ontology_graph:
+                logging.debug(f"Trying to fetch remote triples {uri}.")
+                try:
+                    _g = await fetch_graph(session, uri)
+                    if _g:
+                        self.ontology_graph += _g
```

Comment:
Metoden er litt  misvisende. Den heter fetchTriples men fetcher triple og **setter ontology_graph**. Brukes også for å hente eksterne tripler til data graph. Virker som ortology_graph brukes til flere formål.

Reply:
Renamer til `add_triples` og denne metoden skal kun legge til i `ontology_graph`.

Reply:
Utført i 4e26905